contrib/librdkafka-cmake/CMakeLists.txt

```diff
@@ -112,10 +116,19 @@ if(WITH_SASL_OAUTHBEARER)
   list(APPEND SRCS "${RDKAFKA_SOURCE_DIR}/rdkafka_sasl_oauthbearer.c")
 endif()
 
+## there is a conflict with the cJSON library from aws-s3
+# so normally we link cJSON lirary from contrib/aws (which have some extra patches to be thread safe)
+# but if we don't use aws-s3, we need to link the cJSON library from librdkafka (hacky)
+# which is (potentially) not thread safe (see https://github.com/confluentinc/librdkafka/issues/4159 ),
+if(NOT ENABLE_AWS_S3)
+   list(APPEND SRCS "${RDKAFKA_SOURCE_DIR}/cJSON.c")
+endif()
+
 add_library(_rdkafka ${SRCS})
 add_library(ch_contrib::rdkafka ALIAS _rdkafka)
-
 target_compile_options(_rdkafka PRIVATE -fno-sanitize=undefined)
+target_compile_definitions(_rdkafka PRIVATE -DCJSON_HIDE_SYMBOLS)
```

Comment:
Not sure if you tried that out but what about also hiding the symbols in the aws library?

Reply:
It seems they anyway are combined into one huge link unit, no matter if they private or not....